envy.sh

```diff
@@ -6,58 +6,64 @@ set -euo pipefail
 OUTPUT="${2:-bash}"
 ENVY_NAMESPACE="__ENVY_"
 
-# Taken from https://github.com/jwerle/mush
-mush () {
-  local SELF="$0"
-  local NULL=/dev/null
-  local STDIN=0
-  local STDOUT=1
-  local STDERR=2
-  local LEFT_DELIM="{{"
-  local RIGHT_DELIM="}}"
-  local INDENT_LEVEL="  "
-  local ENV="`env`"
-  local out=">&$STDOUT"
+template () {
+    local RESULT=""
+    local LAST_CHAR=""
 
-  ## read each line
-  while IFS= read -r line; do
-    printf '%q\n' "${line}" | {
-        ## read each ENV variable
-        echo "$ENV" | {
-          while read var; do
-            ## split each ENV variable by '='
-            ## and parse the line replacing
-            ## occurrence of the key with
-            ## guarded by the values of
-            ## `LEFT_DELIM' and `RIGHT_DELIM'
-            ## with the value of the variable
-            case "$var" in
-              (*"="*)
-                key=${var%%"="*}
-                val=${var#*"="*}
-                ;;
-
-              (*)
-                key=$var
-                val=
-                ;;
-            esac
-
-            line="${line//${LEFT_DELIM}$key${RIGHT_DELIM}/$val}"
-          done
+    # Read input char by char
+    while read -n1 CHAR; do
+        # If found {{
+        if [[ "${CHAR}" == "{" && "${LAST_CHAR}" == "{" ]]; then
+            # Remove extra {
+            RESULT="${RESULT::-1}"
+            local VAR_RESULT=""
```

Comment:
Any reason why we choose prefix VAR_ here for local vars on the inner loop? It's ok, but was a bit confusing first read through

Reply:
Yea it is a bit hard to follow, its an inner loop which is searching for the var name inside the brackets